clang/include/clang/Basic/Attr.td

```diff
@@ -1226,6 +1226,17 @@ def SYCLKernel : InheritableAttr {
   let Documentation = [SYCLKernelDocs];
 }
 
+def SYCLSpecialClass: InheritableAttr {
+  let Spellings = [Clang<"sycl_special_class">];
```

Comment:
```suggestion
  let Spellings = [Clang<"sycl_special_class", /*allowInC =*/0>];
```
No need to add this one to C given that it only applies to C++ record declarations.

Reply:
Since this is an internal attribute, we may want to add `__` to the beginning. @AaronBallman  please advice.

Reply:
Prefixing with `__` won't do a whole lot to keep people from using it. One idea we toyed around with is using a `sycl_detail` attribute scope for implementation-details, but there's not been a decision made there.

Anything we do should probably be done to all our attributes to keep them consistent, so I think we can do it in a follow-up. Because it's clearly documented as being an internal attribute, I'm not worried about a deprecation period for it if we do rename it.